### PR TITLE
update plugin versions and fix groovy failures

### LINKED ISSUE #417: New source jobs not created
Looks like new entries in the rosdistro file don't result in new jobs being created (http://repositories.ros.org/status_page/ros_lunar_default.html?q=RED).
I'm not sure where it came from but it seems to have appeared a few days ago. What changed on the buildfarm (that I know of) since then:
- update of all the jenkins plugins and whitelisted all groovy scripts manually
- #415 
- #416

### PR DESCRIPTION
**Pull request by dirk-thomas**
For now this patch only updates the plugin version.

**But** the plugin configurations still need to be reviewed and updated (especially for `groovy`) in order to be parsable with the new plugin versions.

Fixes #417.

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
Got Lrel_reconfigure-jobs working finally. I'm going to merge this and regenerate / deploy all the jobs and see how it goes

**Comment by dirk-thomas** [comment]
Have you checked the configuration from the other updated plugins? While they might not have such breaking changes they still need to be reviewed / updated.

**Comment by mikaelarguedas** [comment]
agreed, but I won't have time to do this today. If this fix allows us to get the buildfarm up for the weekend I'd be in favor of merging and deploying this and iterate on the rest of the plugins as soon as someone has time to do it

File: ros_buildfarm/templates/snippet/reconfigure_jobs.groovy.em [hunk 1]
```diff
@@ -61,7 +61,7 @@ if (views.size() != @(expected_num_views)) {
     throw new AbortException("Wrong number of view configs")
 }
 
-views.each{
+for (it in views) {
```
Comment (tfoote) [inline ros_buildfarm/templates/snippet/reconfigure_jobs.groovy.em]: It is really odd that the iterator was not resolving. But at least the for loop works.

**Comment by dirk-thomas** [comment]
This can be merged once CI has passed.

But the remaining problems have to be followed up on asap:

* check the configuration of the remaining plugins
* update the buildfarm deployment logic to install the diffutils archive
* document the signatures which require approval (if it can't be automated)

**Comment by mikaelarguedas** [comment]
yes I'm planning on creating an issue for 1 and one issue for 2 and 3 with a gist referring all the approval once the list is complete

**Comment by mikaelarguedas** [comment]
Note: this branch should be kept until all jobs have been reconfigured

**Pull request merged**